optuna/storages/rdb/storage.py

```diff
@@ -514,9 +514,13 @@ def _merge_trials_orm(
             for system_attr in id_to_system_attrs[trial_id]:
                 system_attrs[system_attr.key] = json.loads(system_attr.value_json)
 
-            # TODO(Yanase): Use trial.number after TrialModel.number is added.
-            trial_number = self.get_trial_number_from_id(trial_id)
-            result.append(
+            # `0` is a dummy value.
+            # It will be replaced by a proper value before returned to the caller.
+            #
+            # TODO(ohta): Use trial.number after TrialModel.number is added.
+            trial_number = 0
```

Comment:
I think `-1` is safer than `0`, a valid trial number.

Reply:
That makes sense. Thanks.

Reply:
Fixed: https://github.com/pfnet/optuna/pull/361/commits/859b2d2e7e2d8d8f727e974164e6d1da4ac5843a